yajsapi/rest/payment.ts

```diff
@@ -222,14 +217,15 @@ export class Payment {
     return new Invoice(this._api, invoice_obj);
   }
 
-  incoming_invoices(): AsyncGenerator<Invoice> {
+  incoming_invoices(cancellationToken): AsyncGenerator<Invoice> {
     let ts = dayjs().utc();
     let api = this._api;
     let self = this;
 
     async function* fetch(init_ts: Dayjs) {
       let ts = init_ts;
       while (true) {
+        if(cancellationToken.cancelled) break;
```

Comment:
In `market.ts` you are checking `if (cancellationToken && cancellationToken.cancelled)`.